packages/beacon-node/src/network/peers/peerManager.ts

```diff
@@ -693,6 +704,15 @@ export class PeerManager {
       this.logger.verbose("Failed to send goodbye", {peer: prettyPrintPeerId(peer)}, e as Error);
     } finally {
       await this.disconnect(peer);
+      if (goodbye !== GoodByeReasonCode.TOO_MANY_PEERS) {
+        const coolDownMin = 60;
+        // prevent automatic/immediate reconnects
+        this.peerRpcScores.applyReconnectionCoolDown(peer, coolDownMin);
```

Comment:
in `PeerManager.heartbeat()` we call this method after reading peer score, now this method applies score again which does not seem right

we should distinguish consumer of this method call based on `goodbye` param here:
- on `BANNED` or `SCORE_TOO_LOW`, do nothing. The scoring system decided the peer score already
- on `TOO_MANY_PEERS`, we call a new method, for example `applyTooManyPeersCoolDown`, let the scoring system decide a `*BEFORE_DECAY_MS` const for it
- on `IRRELEVANT_NETWORK`, we call a new method `applyIrreleventNetwork`

`applyReconnectionCoolDown` should be used for `onLibp2pPeerDisconnect` __inbound only__